Tests/ComposableArchitectureTests/ReducerTests.swift

```diff
@@ -97,16 +171,72 @@ final class ReducerTests: BaseTCATestCase {
     var first = false
     var second = false
 
+    let _store = Store(
+      initialState: State(counter: 0),
+      reducer: CombineReducers {
+        One(effect: { @MainActor in first = true })
+        Two(effect: { @MainActor in second = true })
+      }.onChange(of: \.counter) { _, _, _ in
```

Comment:
The main issue here is that `Store.init(reducer:)` doesn't take a reducer as a builder context, but `CombineReducers` and other generic reducers only really work inside builder contexts in Swift 5.8 due to the new type checking limitations. So if you plan on combining reducers, you must define a custom conformance with a `body` property:

```swift
struct RootReducer: ReducerProtocol {
  var body: some ReducerProtocol<State, Action> {
    CombineReducers {
      // Note: `first` and `second` must be hoisted out globally to reference here
      One(effect: { @MainActor in first = true })
      Two(effect: { @MainActor in second = true })
    }
    .onChange(of: \.counter) { _, _, _ in
      .send(.increment)
    }
  }
}
```

This compiles fine.

So we should probable add a `Store.init` to takes a builder and maybe soft-deprecate the existing initializers. It's something we've talked about but this is a good example of a problem it solves.